jaxrs-api/src/main/java/jakarta/ws/rs/ext/Part.java

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************
+* Copyright (c) 2021 Eclipse Foundation
+*
+* This specification document is made available under the terms
+* of the Eclipse Foundation Specification License v1.0, which is
+* available at https://www.eclipse.org/legal/efsl.php.
+*******************************************************************/
+package jakarta.ws.rs.ext;
+
+import java.io.InputStream;
+import java.util.Optional;
+
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+
+/**
+ * A {@code Part} is one part of a multipart entity. As defined in
+ * <a href="https://tools.ietf.org/html/rfc7578">RFC 7578</a>, a multipart
+ * request or response must have a content type of "multipart/form-data" with a
+ * {@code boundary} parameter indicating where one part ends the next may begin.
+ * <p>
+ * Multipart entities may be received in a resource method as a collection of
```

Comment:
I would expect that the user would get an error because there is no suitable ParamConverter provider to convert the query/matrix param into a `Part`.  Users or vendors could build their own ParamConverter providers to make that work, but it doesn't seem like that would be a very common use case.